tests/lib/tools/fs-state

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: fs-state wait-for-file [filename] [iterations] [sleep-time]"
+    echo "usage: fs-state mock-dir [dirname]"
+    echo "usage: fs-state restore-dir [dirname]"
+    echo "usage: fs-state mock-file [filename]"
+    echo "usage: fs-state restore-file [filename]"
+}
+
+wait_for_file() {
+    local file="$1"
+    local iters="$2"
+    local sleep_time="$3"
+
+    for _ in $(seq "$iters"); do
+        if [ -e "$file" ]; then
+            echo "File exists: $file"
```

Comment:
I would use the tool name here, to ensure that "File exits: foo" cannot be mistaken inside a log file.
```suggestion
            echo "fs-state: file exists: $file"
```